src/compiler/config/validate-hydrated.ts

```diff
@@ -1,9 +1,9 @@
-import { Config, HydratedFlag } from '../../declarations';
+import { HydratedFlag, UnvalidatedConfig } from '../../declarations';
 import { isString } from '@utils';
 
-export const validateHydrated = (config: Config) => {
-  if (config.hydratedFlag === null || config.hydratedFlag === false) {
-    return null;
+export const validateHydrated = (config: UnvalidatedConfig) => {
+  if (config.hydratedFlag === undefined || config.hydratedFlag === null || config.hydratedFlag === false) {
+    return undefined;
```

Comment:
this was returning `null` but `config.hydratedFlag` is typed `undefined | HydratedFlag`, so I changed the early return to `undefined`